### PR TITLE
fix: fixed the flaky e2e tests that failed the most times and refactor

### PR DESCRIPTION
**Pull request by anikdhabal**
## What does this PR do?

Flaky Tests:-

- [x] Update Profile › Newly added secondary email should show as Unverified
- [x] Routing Forms › Seeded Routing Form › Test preview should return correct route
- [x] Popup Tests › should open embed iframe on click - Configured with light theme
- [x] Inline Iframe - Configured with Dark Theme. Do booking and verify that COEP/CORP headers are correctly set
- [x] Reschedule for booking with seats › If rescheduled/cancelled booking with seats it should display the correct number of seats
- [x] Users can impersonate › App Admin can impersonate users with impersonation enabled
- [x] Workflow Tab - Event Type › Check the functionalities of the Workflow Tab › User Workflows › Editing an existing workflow
- [x] pro user -- future › Time slots should be reserved when selected
- [x] hash my url › generate url hash
- [x] Organization › Email matching orgAutoAcceptEmail and a Verified Organization with DNS Setup Done › existing user migrated to an organization
- [x] Signup Flow Test › Email verification sent if enabled
- [x] Organization › Email not matching orgAutoAcceptEmail › nonexisting user invited to a Team inside organization› By email
- [x] Text area min and max characters text › Create a new event
- [x] Routing Forms › Seeded Routing Form › Router URL should work
- [x] Out of office › User can configure booking redirect
- [x] Out of office › User can edit out of office entry
- [x] MEETING_ENDED, MEETING_STARTED › should create/remove scheduledWebhookTriggers for existing bookings

..............And more

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A-I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

File: apps/web/playwright/booking-pages.e2e.ts [hunk 1]
```diff
@@ -295,7 +295,8 @@ testBothFutureAndLegacyRoutes.describe("pro user", () => {
     await pageTwo.goto(initialUrl);
     await pageTwo.waitForURL(initialUrl);
 
-    await pageTwo.waitForSelector('[data-testid="event-type-link"]');
+    await pageTwo.waitForLoadState("networkidle");
```
Comment (zomars) [inline apps/web/playwright/booking-pages.e2e.ts:298]: I'm not a fan of using waitingForLoadState. This means that either the loading strategy is not optimal or we're not using proper selector.

Comment (zomars) [reply apps/web/playwright/booking-pages.e2e.ts:298]: From [the docs](https://playwright.dev/docs/api/class-frame#frame-wait-for-load-state):
 
image

**Review by zomars — DISMISSED** [review]
Left a [statement](https://github.com/calcom/cal.com/pull/16364/files#r1742471511) about using `waitForLoadState("networkidle")`

File: apps/web/playwright/booking-pages.e2e.ts [hunk 2]
```diff
@@ -295,7 +295,8 @@ testBothFutureAndLegacyRoutes.describe("pro user", () => {
...
+    await pageTwo.waitForSelector('[data-testid="event-type-link"]', { state: "visible" });
```
Comment (zomars) [inline apps/web/playwright/booking-pages.e2e.ts:299]: I think state visible is the default. Isn't it?

Also [waitForSelect is discouraged](https://playwright.dev/docs/api/class-elementhandle#element-handle-wait-for-selector):

image

File: apps/web/playwright/booking-pages.e2e.ts [hunk 3]
```diff
@@ -328,7 +329,8 @@ testBothFutureAndLegacyRoutes.describe("pro user", () => {
...
     const eventTypeLinkTwo = pageTwo.locator('[data-testid="event-type-link"]').first();
```
Comment (zomars) [inline apps/web/playwright/booking-pages.e2e.ts:334]:
```suggestion
    const eventTypeLinkTwo = pageTwo.locator('[data-testid="event-type-link"]').first();
    await eventTypeLinkTwo.waitFor();
```

File: apps/web/playwright/booking-pages.e2e.ts (hunk 2 above)
Comment (anikdhabal) [reply apps/web/playwright/booking-pages.e2e.ts:299]: Make sense thought, i will update

File: apps/web/playwright/booking-pages.e2e.ts (hunk 1 above)
Comment (anikdhabal) [reply apps/web/playwright/booking-pages.e2e.ts:298]: It's true for the `networkidle` arguments, but I found `waitingForLoadState` useful in certain conditions in some cases. Let me recheck if we can avoid using it in favor of a perfect selector instead.

File: apps/web/playwright/availability.e2e.ts [hunk 4]
```diff
@@ -85,7 +85,7 @@ test.describe("Availablity", () => {
     await page.locator('[form="availability-form"][type="submit"]').click();
 
     await expect(dateOverrideList).toHaveCount(3);
-    await expect(await page.getByTitle(deleteButtonTitle).isVisible()).toBe(false);
+    await expect(page.getByTitle(deleteButtonTitle)).toBeHidden();
```
Comment (anikdhabal) [inline apps/web/playwright/availability.e2e.ts:88]: Following the best practices. https://playwright.dev/docs/best-practices#dont-use-manual-assertions

**Review by keithwillcode — APPROVED** [review]

File: apps/web/playwright/webhook.e2e.ts [hunk 5]
```diff
@@ -577,9 +577,9 @@ test.describe("MEETING_ENDED, MEETING_STARTED", async () => {
     expect(newMeetingEndedTriggers.length).toBe(0);
 
     // disable webhook
-    await page.click('[data-testid="webhook-switch"]');
-
-    await page.waitForLoadState("networkidle");
+    await page.getByTestId("webhook-switch").click();
+    const response = await page.waitForResponse("/api/trpc/webhook/edit?batch=1");
+    expect(response.status()).toBe(200);
```
Comment (zomars) [inline apps/web/playwright/webhook.e2e.ts:583]: @anikdhabal this is still flaky. [According to the docs](https://playwright.dev/docs/api/class-page#page-wait-for-response) we need to define the waiting promise first, then trigger the action, then await the promise.

image

Comment (anikdhabal) [reply apps/web/playwright/webhook.e2e.ts:583]: Yah that's true I've traced it. I will fix that in my PR

**Pull request merged**